pkg/controllers/migration/virtualmachine.go

```diff
@@ -78,17 +79,24 @@ func RegisterVMImportController(ctx context.Context, vmware migrationController.
 	importVM.OnChange(ctx, "virtualmachine-import-job-change", vmHandler.OnVirtualMachineChange)
 }
 
-func (h *virtualMachineHandler) OnVirtualMachineChange(key string, vm *migration.VirtualMachineImport) (*migration.VirtualMachineImport, error) {
+func (h *virtualMachineHandler) OnVirtualMachineChange(key string, vmObj *migration.VirtualMachineImport) (*migration.VirtualMachineImport, error) {
 
-	if vm == nil || vm.DeletionTimestamp != nil {
-		return vm, nil
+	if vmObj == nil || vmObj.DeletionTimestamp != nil {
+		return nil, nil
 	}
 
+	vm := vmObj.DeepCopy()
 	switch vm.Status.Status {
 	case "": // run preflight checks and make vm ready for import
 		err := h.preFlightChecks(vm)
 		if err != nil {
-			return vm, err
+			if err.Error() == migration.NotValidDNS1123Label {
+				logrus.Errorf("vm migration target %s in VM %s in namespace %s is not RFC 1123 compliant", vm.Spec.VirtualMachineName, vm.Name, vm.Namespace)
+				vm.Status.Status = migration.VirtualMachineInvalid
+				h.importVM.UpdateStatus(vm)
```

Comment:
would be better to handle it via the [webhook](https://github.com/harvester/harvester/tree/master/pkg/webhook/resources) and better to add exist VM name check as well.

Reply:
we could do that in future releases. I am not sure i would like to change too much so close to RC3.

Reply:
Also realized the validating webhook would need to exist within the vm-import-controller since it is now an optional addon and not deployed by the main harvester chart.

Reply:
Sure, let's leave it as a future enhancement. additionally, CRDs can be created on the Harvester via [here](https://github.com/harvester/harvester/blob/master/pkg/data/crd.go#L18) if needed.